src/providers/xyz.js

```diff
@@ -43,8 +45,22 @@ XYZProvider.prototype = {
     d3.json(url, callback)
   },
 
-  invalidateCache: function () {
-    this.tileCache = {}
+  setReady: function () {
+    this.ready = true
+    this.layer.fire('ready')
```

Comment:
At the moment providers don't have events, but that makes a lot more sense than them knowing about layers.